cpp/src/parquet/column_writer.cc

```diff
@@ -1631,6 +1694,12 @@ class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<
       page_statistics_->UpdateSpaced(values, valid_bits, valid_bits_offset,
                                      num_spaced_values, num_values, num_nulls);
     }
+    if constexpr (std::is_same_v<T, ByteArray>) {
+      if (page_size_stats_builder_ != nullptr) {
+        page_size_stats_builder_->WriteValuesSpaced(values, valid_bits, valid_bits_offset,
```

Comment:
I wonder if we could somehow gather this at a lower level (based on buffer size of written values instead of having to handle Spaced values separately)

Reply:
I have thought about this. We have different interfaces to write values of BYTE_ARRAY type:
- dense `ByteArray` values
- spaced `ByteArray` values
- `arrow::Array` of `String`, `Binary`, and their large variants
- dictionary-encoded `arrow::Array`

These interfaces then directly put values into encoders. So here is the last chance to catch BYTE_ARRAY values before encoding.